components/uniset/src/uniset.rs

```diff
@@ -13,7 +13,8 @@ const BMP_MAX: u32 = 0xFFFF;
 ///
 /// Provides exposure to membership functions and constructors from serialized [`UnicodeSets`](UnicodeSet)
 /// and predefined ranges.
-#[derive(Debug, PartialEq, Hash, Eq)]
+#[derive(Debug, PartialEq, Hash, Eq, Clone)]
+#[cfg_attr(feature = "provider_serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Issue: You still need to validate that the inversion list is valid (sorted, in range, etc).  I think you either need a custom serde::Deserialize impl, or perhaps there's a way to get it via `deserialize_with`.

Reply:
Excluding serde consideration from this particular PR according to [meeting discussion](https://github.com/unicode-org/icu4x/pull/677#issuecomment-840727188).

Reply:
This is not resolved yet.

Reply:
I thought it would be easier (less work) to try to implement the custom deserialization function with an invariant check than it would be to wipe clean the field type change + serde attempt. 

I took the approach of `deserialize_with`, and I merged in the latest from main.

Does that look right? If so, I probably need a negative test for deserializing invalid data, huh?